### PR TITLE
Upgrade NuGet dependencies

### PR DESCRIPTION
**Pull request by pepone**
This PR upgrades NuGet dependencies.

### CODE REVIEW

File: .github/actions/test/action.yaml [hunk 1]
```diff
@@ -7,5 +7,5 @@ runs:
     run: cargo test --manifest-path tools/slicec-cs/Cargo.toml
     shell: bash
   - name: 🧪 Test
-    run: dotnet test --no-build --verbosity normal
+    run: dotnet test --no-build --verbosity normal --blame-hang-timeout 8m
```
Comment (pepone) [inline .github/actions/test/action.yaml:10]: `--blame-hang-timeout` is used to terminate the tests and dump core after the timeout, the NUnit timeout attribute is now deprecated.

File: tests/IceRpc.Tests.Common/AssertTraceListener.cs [hunk 2]
```diff
@@ -12,18 +12,18 @@ public class AssertTraceListener : DefaultTraceListener
...
     {
         if (detailMessage is null || detailMessage.Length == 0)
         {
-            Assert.Fail(message);
+            Assert.Fail(message ?? "");
         }
         else
         {
-            Assert.Fail(message, detailMessage);
+            Assert.Fail($"{message}\n{detailMessage}");
```
Comment (pepone) [inline tests/IceRpc.Tests.Common/AssertTraceListener.cs:26]: The overload with two parameters was removed.

File: tests/IceRpc.Compressor.Tests/AssemblyInfo.cs [hunk 3]
```diff
@@ -1,3 +0,0 @@
-// Copyright (c) ZeroC, Inc.
-
-[assembly: NUnit.Framework.Timeout(8000)]
```
Comment (bernardnormier) [inline tests/IceRpc.Compressor.Tests/AssemblyInfo.cs:3]: When I run "dotnet test", I want a test that takes more than 8s to fail. This shouldn't be a CI-only behavior.

This also seems unrelated to this dependency upgrade.

**Review by externl — APPROVED** [review]

**Review by bernardnormier — APPROVED** [review]
Should we also update Dunet?

File: tests/IceRpc.Compressor.Tests/AssemblyInfo.cs (hunk 3 above)
Comment (pepone) [reply tests/IceRpc.Compressor.Tests/AssemblyInfo.cs:3]: See also https://docs.nunit.org/articles/nunit/writing-tests/attributes/cancelafter.html

Timeout is deprecated, and the new CancelAfter requires to upgrade all test to deal with an extra cancelation token.

When not using CI we can just attach a debugger to the hanging test

**Pull request merged**